azurerm/internal/services/powerbidedicated/validation_test.go

```diff
@@ -0,0 +1,119 @@
+package powerbidedicated
+
+import "testing"
+
+func TestValidateCapacityName(t *testing.T) {
+	testData := []struct {
+		input    string
+		expected bool
+	}{
+		{
+			// empty
+			input:    "",
+			expected: false,
+		},
+		{
+			// basic example
+			input:    "hello",
+			expected: true,
+		},
+		{
+			// can't start with an underscore
+			input:    "_hello",
+			expected: false,
+		},
+		{
+			// can't end with a dash
+			input:    "hello-",
+			expected: false,
+		},
+		{
+			// can't contain an exclamation mark
+			input:    "hello!",
+			expected: false,
+		},
+		{
+			// can't contain dash in the middle
+			input:    "malcolm-in-the-middle",
+			expected: false,
+		},
+		{
+			// can't end with a period
+			input:    "hello.",
+			expected: false,
+		},
+		{
+			// 63 chars
+			input:    "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijk",
+			expected: true,
+		},
+		{
+			// 64 chars
+			input:    "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkj",
+			expected: true,
+		},
+		{
+			// 65 chars
+			input:    "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkja",
+			expected: false,
+		},
+	}
+
+	for _, v := range testData {
+		t.Logf("[DEBUG] Testing %q..", v.input)
+
+		_, errors := ValidateCapacityName(v.input, "name")
+		actual := len(errors) == 0
+		if v.expected != actual {
+			t.Fatalf("Expected %t but got %t", v.expected, actual)
+		}
+	}
+}
+
+func TestValidateCapacityAdministratorName(t *testing.T) {
+	testData := []struct {
+		input    string
+		expected bool
+	}{
+		{
+			// empty
+			input:    "",
+			expected: false,
+		},
+		{
+			// basic example
+			input:    "hello",
+			expected: false,
+		},
+		{
+			// valid email address
+			input:    "hello@microsoft.com",
+			expected: true,
+		},
+		{
+			// invalid email address
+			input:    "hello@microsoft",
```

Comment:
this is a valid email address - so should be allowed - please see https://tools.ietf.org/html/rfc2822

Reply:
fixed